hal/src/thumbv7em/clock/v2/types.rs

```diff
@@ -1,256 +1,102 @@
-//! # Clocking API v2 type infrastructure
```

Comment:
Do we want to totally scrap it? It obviously does not have to be here or even in that form but some kind of design document giving an overview of design choices would be nice for future maintainers or even us if we forget how some parts work :D